packages/@sanity/cli/src/actions/init-project/initProject.ts

```diff
@@ -752,6 +782,125 @@ export default async function initSanity(
+        print(
+          `\nDetected framework ${chalk.blue(framework?.name)}, using prefix '${
+            framework.envPrefix
+          }'`
+        )
+      }
+
+      await writeEnvVarsToFile(filename, envVars, {
+        framework,
+        outputPath,
+      })
+    } catch (err) {
+      print(err)
+      throw new Error('An error occurred while creating .env', {cause: err})
+    }
+  }
+
+  async function writeEnvVarsToFile(
+    filename: string,
+    envVars: Record<string, string>,
+    options: {framework: Framework | null; outputPath: string}
+  ) {
+    const envPrefix = options.framework?.envPrefix || ''
+    const keyPrefix = envPrefix.includes('SANITY') ? envPrefix : `${envPrefix}SANITY_`
+    const fileOutputPath = path.join(options.outputPath, filename)
+
+    // prepend framework and sanity prefix to envVars
+    for (const key of Object.keys(envVars)) {
+      envVars[`${keyPrefix}${key}`] = envVars[key]
+      delete envVars[key]
+    }
+
+    // make folder if not exists (if output path is specified)
+    try {
+      await fs.mkdir(options.outputPath, {recursive: true})
+    } catch (err) {
+      debug('Error creating folder %s', options.outputPath)
+    }
+
+    // time to update or create the file
+    try {
+      const existingEnv = await fs.readFile(fileOutputPath, {encoding: 'utf8'})
+      const updatedEnv = parseAndUpdateEnvVars(existingEnv, envVars)
+      await fs.writeFile(fileOutputPath, updatedEnv, {
+        encoding: 'utf8',
+      })
+    } catch (err) {
+      await fs.writeFile(
+        fileOutputPath,
+        Object.keys(envVars)
+          .map((key) => `${key}="${envVars[key]}"`)
+          .join('\n')
+          .concat('\n'),
+        {
+          encoding: 'utf8',
+        }
+      )
+    }
+
+    print(`\n${chalk.green('Success!')} Environment variables written to ${fileOutputPath}`)
+  }
+
+  function parseAndUpdateEnvVars(fileContents: string, envVars: Record<string, string>): string {
+    const existingKeys = dotenv.parse(fileContents) // this will contain all vars
+    const updatedKeys: Record<string, string> = {} // this will only contain our vars
+
+    // find and update keys
+    for (const [key, value] of Object.entries(envVars)) {
+      if (!existingKeys[key]) {
+        updatedKeys[key] = value
+        print(`Appended ${key}="${envVars[key]}"`)
+        continue
+      }
+
+      print(`Found existing ${key}, replacing value.`)
+      updatedKeys[key] = value
+    }
+
+    // clone fileContents and replace existing keys by string matching
+    let updatedEnv = fileContents.repeat(1)
+    for (const [key, value] of Object.entries(updatedKeys)) {
+      if (existingKeys[key]) {
+        const existingValue = existingKeys[key]
+
+        updatedEnv = updatedEnv
+          .split('\n')
+          .map((line) => {
+            if (
+              !line.trim().startsWith('#') && // ignore comments
+              new RegExp(`(^\s?${key})[= :]`).test(line) // match key
```

Comment:
```suggestion
              new RegExp(`(^\\s*${key})((: )|( *=))`).test(line) // match key
```

I assume this was meant to be "one whitespace character, or none, at the start of the line, followed by the key, followed by a colon/equal sign. However, the current regex says:

One or none `s` character at the start of the line, followed by the key, followed by either an equal sign, a space or a colon.

My suggestion (if I got it right) says "any number of whitespace characters at the start of the line, followed by the key, followed by either a colon and a space, OR any number of spaces followed by an equals sign". The colon/equal rules and spaces I pulled together from a brief test of the dotenv parser: the colon syntax does not support spaces after the key, whereas the equal syntax does.

Whether or not you accept my suggestion, the `s` character seems like a bug - I'm sure it was just one too few escapes.

Reply:
good catch! :)) i don't speak regex too well, but ran this through some manual testing and it seems to work as good as before 👍🏼

Reply:
Oh, that `\s?` can probably be blamed on me from an earlier suggestion 🙈